x-pack/plugins/lens/public/indexpattern_datasource/indexpattern.tsx

```diff
@@ -468,6 +468,16 @@ export function getIndexPatternDatasource({
       const ids = Object.values(state.layers || {}).map(({ indexPatternId }) => indexPatternId);
       return ids.filter((id) => !state.indexPatterns[id]);
     },
+    isTimeBased: (state) => {
+      const { layers } = state;
+      return (
+        Boolean(layers) &&
+        Object.values(layers).some((layer) => {
+          const buckets = layer.columnOrder.filter((colId) => layer.columns[colId].isBucketed);
+          return buckets.some((colId) => layer.columns[colId].operationType === 'date_histogram');
```

Comment:
nit: let's replace  'date_histogram' -> BUCKET_TYPES.DATE_HISTOGRAM from the data plugin

Reply:
For now I'll leave it as it is as it is spread like this across all the files in Lens... Maybe we could address it at once in separate PR.